commit 51224f8a71b45245dfbbe0e467c5dd6c2c012ac2
Author: Zdenek Pavlas <zpavlas@redhat.com>
Date:   Fri Oct 18 11:01:42 2013 +0200

    mergerepo: merge package versions from multiple repos. BZ 1020756

diff --git a/createrepo/merge.py b/createrepo/merge.py
index 1ac43bb..80d9df5 100644
--- a/createrepo/merge.py
+++ b/createrepo/merge.py
@@ -70,9 +70,7 @@ class RepoMergeBase:
 
         for repo in repos:
             for pkg in repo.sack:
-                others = self.yumbase.pkgSack.searchNevra(name=pkg.name, arch=pkg.arch)
-                # NOTE the above is definitely going to catch other versions which may
-                # be an invalid comparison
+                others = self.yumbase.pkgSack.searchNevra(pkg.name, pkg.epoch, pkg.version, pkg.release, pkg.arch)
                 if len(others) > 1:
                     for thatpkg in others:
                         if pkg.repoid == thatpkg.repoid: continue
