diff -up ./createrepo/__init__.py.original ./createrepo/__init__.py
--- ./createrepo/__init__.py.original	2013-11-28 16:15:38.574519868 +0900
+++ ./createrepo/__init__.py	2013-11-28 16:15:41.223494682 +0900
@@ -230,6 +230,7 @@ class MetaDataGenerator:
                 except (OSError, IOError), e:
                     raise MDError, _("Could not create exclusive lock in %s and sqlite database generation enabled. Is this path on nfs? Is your lockd running?") % direc
                 else:
+                    f.close()
                     os.unlink(direc + '/locktest')
                 
         if self.conf.deltas:
