diff -up createrepo-0.9.9/genpkgmetadata.py.old createrepo-0.9.9/genpkgmetadata.py
--- createrepo-0.9.9/genpkgmetadata.py.old	2018-04-24 15:53:12.952201134 +0200
+++ createrepo-0.9.9/genpkgmetadata.py	2018-04-24 15:54:31.596657882 +0200
@@ -284,6 +284,8 @@ def main(args):
     except MDError, errormsg:
         errorprint(_('%s') % errormsg)
         # cleanup
+        if conf.outputdir is None:
+            conf.outputdir = ''
         tmp = os.path.join(conf.outputdir, conf.tempdir)
         if os.path.exists(tmp):
             for name in os.listdir(tmp):
