diff -up createrepo-0.9.9/createrepo/__init__.py.old createrepo-0.9.9/createrepo/__init__.py
--- createrepo-0.9.9/createrepo/__init__.py.old	2017-06-28 13:35:42.857445052 +0200
+++ createrepo-0.9.9/createrepo/__init__.py	2017-06-28 13:37:07.825166534 +0200
@@ -1209,7 +1209,7 @@ class MetaDataGenerator:
 
 
     def _cleanup_update_tmp_dir(self):
-        if self.conf.update:
+        if self.conf.update and hasattr(self, 'oldData'):
             self.oldData.cleanup()
     
         
