--- a/createrepo/__init__.py    2015-08-24 14:58:31.510985403 +0900
+++ b/createrepo/__init__.py    2015-08-24 14:58:52.842172340 +0900
@@ -230,6 +230,7 @@ class MetaDataGenerator:
                 except (OSError, IOError), e:
                     raise MDError, _("Could not create exclusive lock in %s and sqlite database generation enabled. Is this path on nfs? Is your lockd running?") % direc
                 else:
+                    f.close()
                     os.unlink(direc + '/locktest')
                 
         if self.conf.deltas:
