commit f0020c470935cbb98ed4c53ef8ae24952e799a01
Author: Zdeněk Pavlas <zpavlas@redhat.com>
Date:   Fri Dec 14 11:20:21 2012 +0100

    no repomd.xml && --checkts: skip .rpm timestamp checking.  BZ 877301

diff --git a/createrepo/__init__.py b/createrepo/__init__.py
index a52ec9e..1b18a9f 100644
--- a/createrepo/__init__.py
+++ b/createrepo/__init__.py
@@ -327,7 +327,7 @@ class MetaDataGenerator:
     def checkTimeStamps(self):
         """check the timestamp of our target dir. If it is not newer than
            the repodata return False, else True"""
-        if self.conf.checkts:
+        if self.conf.checkts and self.conf.mdtimestamp:
             dn = os.path.join(self.conf.basedir, self.conf.directory)
             files = self.getFileList(dn, '.rpm')
             files = self.trimRpms(files)
